File: Samples/CSharp/ImagingDemo1/Form1.cs
//----------------------------------------------------------------------------
//  This file is part of the OpenNETCF Smart Device Framework Code Samples.
// 
//  Copyright (C) OpenNETCF Consulting, LLC.  All rights reserved.
// 
//  This source code is intended only as a supplement to Smart Device 
//  Framework and/or on-line documentation.  
// 
//  THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//  PARTICULAR PURPOSE.
//----------------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.ComponentModel;
using Microsoft.WindowsMobile.Forms;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
//
using OpenNETCF.Drawing.Imaging;

namespace ImagingDemo1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void mnuLoadDirect_Click(object sender, EventArgs e)
        {
            SelectPictureDialog dlg = new SelectPictureDialog();
            if (dlg.ShowDialog() == DialogResult.OK)
            {
                try
                {
                    pbImage.Image = new Bitmap(dlg.FileName);
                }
                catch (OutOfMemoryException)
                {
                    MessageBox.Show("Out of memory");
                }
            }
        }

        private void mnuLoadImaging_Click(object sender, EventArgs e)
        {
            SelectPictureDialog dlg = new SelectPictureDialog();
            if (dlg.ShowDialog() == DialogResult.OK)
            {
                try
                {
                    ImagingFactory factory = new ImagingFactoryClass();
                    IImage img;
                    factory.CreateImageFromFile(dlg.FileName, out img);
                    IBitmapImage imgB;
                    factory.CreateBitmapFromImage(img, 
                        (uint)pbImage.Width, 
                        (uint)pbImage.Height, 
                        System.Drawing.Imaging.PixelFormat.Format24bppRgb, 
                        InterpolationHint.InterpolationHintDefault, 
                        out imgB);
                    pbImage.Image = ImageUtils.IBitmapImageToBitmap(imgB);
                }
                catch (OutOfMemoryException)
                {
                    MessageBox.Show("Out of memory");
                }
            }
        }
    }
}

File: Samples/VB/ImagingDemo1/Form1.vb
Imports Microsoft.WindowsMobile.Forms
Imports OpenNETCF.Drawing.Imaging

Public Class Form1

    Private Sub mnuLoadImaging_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Dim dlg As New SelectPictureDialog()
        If dlg.ShowDialog() = DialogResult.OK Then
            Try
                Dim factory As ImagingFactory = New ImagingFactoryClass()
                Dim img As IImage
                factory.CreateImageFromFile(dlg.FileName, img)
                Dim imgB As IBitmapImage
                factory.CreateBitmapFromImage(img, CUInt(pbImage.Width), CUInt(pbImage.Height), System.Drawing.Imaging.PixelFormat.Format24bppRgb, InterpolationHint.InterpolationHintDefault, imgB)
                pbImage.Image = ImageUtils.IBitmapImageToBitmap(imgB)
            Catch generatedExceptionName As OutOfMemoryException
                MessageBox.Show("Out of memory")
            End Try
        End If

    End Sub
    Private Sub mnuLoadDirect_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Dim dlg As SelectPictureDialog = New SelectPictureDialog()
        If dlg.ShowDialog() = DialogResult.OK Then
            Try
                pbImage.Image = New Bitmap(dlg.FileName)
            Catch
                MessageBox.Show("Out of memory")
            End Try
        End If
    End Sub
End Class
